controller/rebuild.go

```diff
@@ -76,17 +91,29 @@ func (c *Controller) VerifyRebuildReplica(address string) error {
 		return err
 	}
 
-	logrus.Infof("chain %v from rw replica %s", rwChain, rwReplica.Address)
-	// Don't need to compare the volume head disk
-	rwChain = rwChain[1:]
-
 	chain, err := getReplicaChain(address)
 	if err != nil {
 		return err
 	}
+	WOcheckpoint, err := getReplicaCheckpoint(address)
+	if err != nil {
+		return err
+	}
+	var (
+		indx     int
+		snapshot string
+	)
+	for indx, snapshot = range rwChain {
+		if snapshot == WOcheckpoint {
+			break
+		}
+	}
+	logrus.Infof("chain %v from rw replica %s", rwChain, rwReplica.Address)
+	// No need to compare the volume head disk
+	rwChain = rwChain[1 : indx+1]
```

Comment:
WOCheckPoint can be empty also, and so, indx + 1 can lead to out of boundary

Reply:
in rwChain[1:indx+1], indx+1 element is not included. It is similar to [1:indx+1).

Reply:
https://play.golang.org/p/JhEmlx8FHMb